### ISSUE TITLE
File.getcwd results differently on Windows

### ISSUE THREAD

**Issue by elgiano** [supercollider/supercollider#5757]
## Environment

* SuperCollider version: 3.12.2
* Operating system: Windows

## Steps to reproduce

```supercollider
File.getcwd
```

## Expected vs. actual behavior

On Linux, `File.getcwd` gives the user's home directory. On Windows it gives the script's working directory, like `thisProcess.nowExecutingPath.dirname`. This difference can be confusing and create portability problems, for example when writing SC code on windows and then deploying it on a Raspberry Pi, and I couldn't find references to it in the docs.

I faced this problem while working with FileReader.

**Comment by elgiano** [comment]
On a second thought, the problem was that the user opened a `.scd` file from a folder, and thus launched `scide` from that folder.
It then made perfect sense that `File.getcwd` gave that folder, I had misinterpreted what was actually correct and consistent behavior.

However, maybe a note could be added to the docs about the importance of using absolute paths to avoid running in this kind of confusion?

**Comment by jamshark70** [comment]
I think the doc change should just be to explain how the current working directory is determined.

This is the sum total of `getcwd`'s documentation: "POSIX standard 'get current working directory'" -- which assumes rather a lot of foreknowledge.

**Comment by dyfer** [comment]
Thanks @elgiano , I think it would be good to add this. I have two suggestions:
- In "See also" I'd link both `Process -nowExecutingPath` and `String -resolveRelative`
- I'd add the examples for nowExecutingPath and resolveRelative in respective helpfiles, not in `getcwd`
  - in "Process": `thisProcess.nowExecutingPath.dirname;`
  - in "String": `"".resolveRelative;` (I use this one more often as it's shorter and I can remember it easier)

**Comment by elgiano** [comment]
Hey @dyfer!
I agree to add a "See also" reference to `resolveRelative`, I actually didn't know this function!
About the example, the idea was to put `nowExecutingPath` next to `getcwd` to support clarifying their difference, since there can be confusion.

Now I think that at least these functions should all refer to each other: `File -getcwd`, `Process -nowExecutingPath`, `String -resolveRelative` and maybe also `Document.current.dir`? Then IMO one of them should have an example that compares them, but I'm now confused about which one should it be. What do you think?

**Comment by dyfer** [comment]
I agree that adding cross-references to all of them would be great.

> Now I think that at least these functions should all refer to each other: `File -getcwd`, `Process -nowExecutingPath`, `String -resolveRelative` and maybe also `Document.current.dir`? Then IMO one of them should have an example that compares them, but I'm now confused about which one should it be. What do you think?

I'd either leave the single examples with each respective class, or... At the expense of slight code duplication, but gaining better visibility, what do you think about having all 4 examples in all 4 places? One line each, like:
```
"".resolveRelative; // returns a directory of the currently open scd file
thisProcess.nowExecutingPath.dirname;  // returns a directory of the currently open scd file
File -getcwd; // ....
Document.current.dir; // ...
```
Either that, or leave each example with the respective classes and hope that cross-links are clear enough.

**Comment by prko** [comment]
In each platform, `File.getcwd`returns different results:
- on Linux: `~`
- on macOS: `/`
- on Windows: `Platform.resourceDir`
If it is not a bug, it is better to update `File.schelp` only.

But, is this not a bug? Sc should return a consistent result on all platforms.

The related line is hardcoded:

https://github.com/supercollider/supercollider/blob/124b43e84b07c7185d9a080bcf3d77a050f5dab5/SCClassLibrary/Common/Files/File.sc#L90-L94
https://github.com/supercollider/supercollider/blob/124b43e84b07c7185d9a080bcf3d77a050f5dab5/SCClassLibrary/Common/Files/File.sc#L168-L171

Someone good at c++ could resolve it...

**Comment by jamshark70** [comment]
> But, is this not a bug? Sc should return a consistent result on all platforms.

I don't think it needs to be consistent, because the filesystem layouts are very different in the 3 OSes.

File.getcwd should return the sclang process's current working directory. This is set at the commandline, or by the active cwd when launching sclang.

```
$ sclang -d /home/dlm/tmp
... snip...

sc3> File.getcwd
-> /home/dlm/tmp
```

getcwd should not change to override the actual cwd.

So the issue that you're reporting is that the default cwd is not the same in different OSes... which I think is not so weird.

But it may be good to review those defaults -- maybe the defaults could be improved.

**Comment by prko** [comment]
>But it may be good to review those defaults -- maybe the defaults could be improved.

It would be great!

However, if the defaults will be preserved as the current paths, how about adding the following lines in schelp?
```
`File.getcwd` behaves differently depending on the operating system due to differences in filesystem layouts among the three OSes:

- On Linux: Returns the user's home directory (`~`).
- On macOS: Returns the root directory (`/`).
- On Windows: Returns the resource directory (`Platform.resourceDir`).

This behavior reflects the specific conventions and structures of each platform's filesystem.
```

**Comment by capital-G** [comment]
> > But it may be good to review those defaults -- maybe the defaults could be improved.
> 
> It would be great!
> 
> However, if the defaults will be preserved as the current paths, how about adding the following lines in schelp?
> 
> ```
> `File.getcwd` behaves differently depending on the operating system due to differences in filesystem layouts among the three OSes:
> 
> - On Linux: Returns the user's home directory (`~`).
> - On macOS: Returns the root directory (`/`).
> - On Windows: Returns the resource directory (`Platform.resourceDir`).
> 
> This behavior reflects the specific conventions and structures of each platform's filesystem.
> ```

As mentioned it mostly depends on how sclang is launched.
Your mentioned values seem to be the default values from QtIDE and they can be specified/changed by the user.
By using another IDE (or plainly using CLI) these values can differ.

<img width="568" alt="Image" src="https://github.com/user-attachments/assets/0ae9b84d-721d-41df-b129-5341b7c2d0e1" />

https://github.com/supercollider/supercollider/blob/124b43e84b07c7185d9a080bcf3d77a050f5dab5/editors/sc-ide/core/sc_process.cpp#L121-L130

I currently can't find the section where the default values of these settings origin from.


Personally I don't think we need to document the values in this case (simply evaluate them) but we should instead document how these can be set (using CLI/IDE).
